File: tlg/src/ch/bailu/tlg/TlgPoint.java
package ch.bailu.tlg;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.IOException;

public class TlgPoint {
    public int x,y;
    
    public TlgPoint() {};
    
    public TlgPoint(int x, int y) {
        this.x=x; 
        this.y=y;
    }

    public TlgPoint(BufferedInputStream reader) throws IOException {
        
        x=ByteInteger.read(reader);
        y=ByteInteger.read(reader);
    }
    
    public void writeState(BufferedOutputStream writer) throws IOException {
        ByteInteger.wrap(x).writeState(writer);
        ByteInteger.wrap(y).writeState(writer);
    }
    public TlgPoint(TlgPoint p) {
        x=p.x; y=p.y;
    }
}

File: tlg/src/main/kotlin/tlg/geometry/TlgPoint.kt
package tlg.geometry

import tlg.context.ByteInteger
import java.io.BufferedInputStream
import java.io.BufferedOutputStream
import java.io.IOException

class TlgPoint {
    var x: Int
    var y: Int

    constructor(x: Int = 0, y: Int = 0) {
        this.x = x
        this.y = y
    }

    constructor(p: TlgPoint) {
        x = p.x
        y = p.y
    }

    constructor(reader: BufferedInputStream) {
        x = ByteInteger.read(reader)
        y = ByteInteger.read(reader)
    }

    @Throws(IOException::class)
    fun writeState(writer: BufferedOutputStream) {
        ByteInteger.wrap(x).writeState(writer)
        ByteInteger.wrap(y).writeState(writer)
    }
}
